File: Dynamic_Programming/2/dp.cpp
#include <iostream>
using namespace std;

const int N = 1010;
int n, m, v[N], w[N], f[N];

int main() {
    scanf("%d%d", &n, &m);
    for (int i = 1; i <= n; i++) scanf("%d%d", &v[i], &w[i]);
    for (int i = 1; i <= n; i++) {
        for (int j = m; j >= v[i]; j--) {
            f[j] = max(f[j], f[j - v[i]] + w[i]);
        }
    }
    printf("%d", f[m]);
    return 0;
}

File: Dynamic_Programming/2/dp.py
if __name__ == "__main__":
    n, m = map(int, input().split())
    dp = [0] * (m + 1)
    for _ in range(n):
        v, w = map(int, input().split())
        for j in range(m, v - 1, -1):
            dp[j] = max(dp[j], dp[j - v] + w)
    print(dp[m])
